store/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonWriterBuilder.java

```diff
@@ -416,16 +411,58 @@ private CarbonTable buildCarbonTable() {
     }
     TableSchema schema = tableSchemaBuilder.build();
     schema.setTableName(tableName);
-    CarbonTable table = CarbonTable.builder()
-        .tableName(schema.getTableName())
-        .databaseName(dbName)
-        .tablePath(path)
-        .tableSchema(schema)
-        .isTransactionalTable(isTransactionalTable)
-        .build();
+    CarbonTable table =
+        CarbonTable.builder().tableName(schema.getTableName()).databaseName(dbName).tablePath(path)
+            .tableSchema(schema).isTransactionalTable(isTransactionalTable).build();
     return table;
   }
 
+  private void buildTableSchema(Field[] fields, TableSchemaBuilder tableSchemaBuilder,
+      List<String> sortColumnsList, ColumnSchema[] sortColumnsSchemaList) {
+    for (Field field : fields) {
+      if (null != field) {
+        int isSortColumn = sortColumnsList.indexOf(field.getFieldName());
+        if (isSortColumn > -1) {
+          // unsupported types for ("array", "struct", "double", "float", "decimal")
+          if (field.getDataType() == DataTypes.DOUBLE || field.getDataType() == DataTypes.FLOAT
+              || DataTypes.isDecimal(field.getDataType()) || DataTypes
+              .isArrayType(field.getDataType()) || DataTypes.isStructType(field.getDataType())) {
```

Comment:
use field.getDataType().isComplexType() to check for complex types

Reply:
Done